packages/protocol/contracts/tokenvault/BridgedERC20.sol

```diff
@@ -65,13 +73,21 @@ contract BridgedERC20 is BridgedERC20Base, IERC20MetadataUpgradeable, ERC20Upgra
         // Initialize OwnerUUPSUpgradable and ERC20Upgradeable
         __Essential_init(_addressManager);
         __ERC20_init({ name_: _name, symbol_: _symbol });
+        __ERC20Snapshot_init();
+        __ERC20Votes_init();
+        __ERC20Permit_init(_name);
 
         // Set contract properties
         srcToken = _srcToken;
         srcChainId = _srcChainId;
         srcDecimals = _decimals;
     }
 
+    /// @notice Creates a new token snapshot.
+    function snapshot() public onlyOwner {
```

Comment:
By default the owner of this smart contract is also the owner of the vault. So will we have a system to give token projects ownership of the bridged contract or snapshots will simply only be used for TKO?

Reply:
We cannot transfer the ownership to some else, so we need another role for snapshot function.